### ISSUE TITLE
Application Domain Model

### ISSUE THREAD

**Issue by nasadorian** [jorgecotillo/green_light_health#23]
As a developer, 

I need a high level domain model describing the entities and transactions involved in the application. The domain model should capture all of the information flows happening from the perspectives of different users, and inform the design and development of concrete aspects of the software system. Some examples of entities and transactions in the domain model entities are listed below for discussion.

* User
  - An entity in the application, having a number of "flavors"
  - Business Owner, End User, Employee, etc

* Business
  - An entity representing a business which has opted into the contact tracing application
  - A business may have many Business Locations

* Health Status Code
  - A state in the application which can be applied to a User or a Business Location
  - Green: No known exposures
  - Yellow: A user which may have been exposed to a pathogen (transitive) has visited the Business Location
  - Red: A user visiting the Business Location has shown symptoms of the pathogen

* Business Location
  - An entity representing a specific geographic location of a business
  - Has geographic coordinates, an address, capacity information, indoor/outdoor, etc
  - Is subject to a Health Status Code depending Users who have visited and their own statuses
  - A User must present their Health Status Code when visiting the Business Location

* Health Declaration
  - A document/questionnaire which the User must answer within the 24 hours they are visiting a location
  - Describes symptoms of the pathogen and requires the User declare whether or not they have felt them
  - A User will generate many Health Declarations over their lifetime in the application

**Comment by noweber** [comment]
@nasadorian I don't agree with this user story being on the backlog. I agree that modeling the domain model can have value to the developers, but this user story seems to suggest solving the long term vision of the product up front. We should instead work through more granular and urgent problems gathering feedback along the way from our stakeholders to keep adjusting to theirs wants/needs. We should only model and design as much as is needed to solve the most urgent problems and allow it to become a continuous conversation among the development team. As such, each of these pieces would then become a part of solving the existing user stories and problems our users face. I recommend we embed this into those vertical slices.

**Comment by noweber** [comment]
@nasadorian To clarify, I think these should be tasks underneath multiple user stories rather than a single user story itself.
